app/src/main/java/ca/marshallasch/veil/database/Database.java

```diff
@@ -1463,4 +1535,88 @@ public void clearSyncStats() {
     public void clearPeers() {
         getWritableDatabase().delete(SyncStatsEntry.TABLE_NAME, null, null);
     }
+
+    /**
+     * This function removes all the posts and comments from the database that are
+     * not flagged by the protecton
+     */
+    @WorkerThread
+    public void dataSaverClear(){
+        String whereClause = "status = "+KnownPostsContract.POST_NORMAL;
+        getReadableDatabase().delete(KnownPostsEntry.TABLE_NAME, whereClause, null);
+    }
+
+    /**
+     * Gets all the posts hashes based on their status
+     * @param _status indicates status 1,2, or 3
+     *               where
+     *               1 = normal
+     *               2 = protected
+     *               3 = dead
+     */
+    public List<String> getAllPostsbyStatus(int _status){
```

Comment:
This will give far too many post hashes for normal posts. it will contain a lot of duplicates and will return more rows then needed because it is not just looking at the rows where there is no comment hash.

Reply:
changed function to return distinct hashes only + capitalized 'b'